### PR TITLE
[PP-28] Update application structure

### PR DESCRIPTION
**Pull request by MakFill**
Task: https://andrei-lysiuk.atlassian.net/browse/PP-28

### CODE REVIEW

File: src/helpers/getEnvValue.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export function getEnvValue(): string {
```
Comment (mrakofff) [inline src/helpers/getEnvValue.ts]: this function should return `env` variable by provided `env` name. F.e.: `NODE_ENV` -> `process.env.NODE_ENV`

File: src/store/reducers/userReducer.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { createReducer } from '@reduxjs/toolkit';
+import {
+  addGameSettings, addUserId, addUserName, addUserRoom,
+} from '../actions/userActions';
```
Comment (mrakofff) [inline src/store/reducers/userReducer.ts]: use `alias`

File: src/store/reducers/userReducer.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+const initialState = {
+  user: {
```
Comment (mrakofff) [inline src/store/reducers/userReducer.ts]: split to different reducers: `user`, `game`

File: src/store/selectors/userSelectors.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { createSelector } from 'reselect';
+import { RootState } from '../store';
+
+const user = (state: RootState) => state.userReducer.user;
+const settings = (state: RootState) => state.userReducer.gameSettings;
+
+export const userId = createSelector(user, (state) => state.id);
+export const userName = createSelector(user, (state) => state.name);
+export const userRoom = createSelector(user, (state) => state.room);
+export const gameSettings = createSelector(settings, (state) => state);
```
Comment (mrakofff) [inline src/store/selectors/userSelectors.ts]: combine user selectors as `userSelectors` object and make one export to simplify imports

File: src/store/store.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
+import { getEnvValue } from '@/helpers/getEnvValue';
+import { userReducer } from './reducers/userReducer';
+
+export const store = configureStore({
+  reducer: { userReducer },
+  middleware: getDefaultMiddleware({
```
Comment (mrakofff) [inline src/store/store.ts:8]: could be  `middleware: (getDefaultMiddleware) => getDefaultMiddleware(<code>)` but it's okay too

**Review by mrakofff — CHANGES_REQUESTED** [review]

File: src/store/selectors/userSelectors.ts (hunk 4 above)
Comment (MakFill) [reply src/store/selectors/userSelectors.ts]: done

File: src/store/reducers/userReducer.ts (hunk 3 above)
Comment (MakFill) [reply src/store/reducers/userReducer.ts]: done

File: src/store/reducers/userReducer.ts (hunk 2 above)
Comment (MakFill) [reply src/store/reducers/userReducer.ts]: done

File: src/helpers/getEnvValue.ts (hunk 1 above)
Comment (MakFill) [reply src/helpers/getEnvValue.ts]: done

File: src/store/reducers/userReducer.ts (hunk 3 above)
Comment (mrakofff) [reply src/store/reducers/userReducer.ts]: now your reducer `userReducer.ts` has 2 reducers inside
and you don't need `user` nesting

File: src/store/selectors/userSelectors.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { createSelector } from '@reduxjs/toolkit';
+import { RootState } from '@/store';
+
+const user = (state: RootState) => state.userReducer.user;
+const settings = (state: RootState) => state.gameReducer.gameSettings;
+
+export const userSelectors = {
+  userId: createSelector(user, (state) => state.id),
```
Comment (mrakofff) [inline src/store/selectors/userSelectors.ts]: if you do not use this selectors do not create them

File: src/store/selectors/userSelectors.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  userName: createSelector(user, (state) => state.name),
+  userRoom: createSelector(user, (state) => state.room),
+  gameSettings: createSelector(settings, (state) => state),
```
Comment (mrakofff) [inline src/store/selectors/userSelectors.ts]: `gameSettings` selector inside `userSelectors`?

File: src/utils/interfaces/index.ts [hunk 8]
```diff
@@ -0,0 +1 @@
+export * from './interfaces';
```
Comment (mrakofff) [inline src/utils/interfaces/index.ts]: each interface should be in separate named file and reexported in `index` file

**Review by mrakofff — CHANGES_REQUESTED** [review]

File: src/store/selectors/userSelectors.ts (hunk 6 above)
Comment (MakFill) [reply src/store/selectors/userSelectors.ts]: delete all selectors?

File: src/utils/interfaces/index.ts (hunk 8 above)
Comment (MakFill) [reply src/utils/interfaces/index.ts]: done

File: src/store/selectors/userSelectors.ts (hunk 7 above)
Comment (MakFill) [reply src/store/selectors/userSelectors.ts]: replaced

File: src/store/selectors/userSelectors.ts (hunk 6 above)
Comment (mrakofff) [reply src/store/selectors/userSelectors.ts]: leave user data selector(`state.user`)

File: src/store/store.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
+import { getEnvValue } from '@/helpers/getEnvValue';
+import { userReducer, gameReducer } from '@/store/reducers/userReducer';
+
+export const store = configureStore({
+  reducer: { userReducer, gameReducer },
```
Comment (mrakofff) [inline src/store/store.ts]: change name without `reducer` postfix
image
https://redux.js.org/style-guide/style-guide#name-state-slices-based-on-the-stored-data

File: src/store/selectors/userSelectors.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+
+const user = (state: RootState) => state.userReducer;
+
+export const userSelectors = {
+  userId: createSelector(user, (state) => state.id),
+  userName: createSelector(user, (state) => state.name),
+  userRoom: createSelector(user, (state) => state.room),
+};
+
+export const gameSettings = createSelector((state: RootState) => state.gameReducer, (state) => state);
```
Comment (mrakofff) [inline src/store/selectors/userSelectors.ts]: split reducers and selectors to diff files according to their names and logic(f.e. user -> user logic, game -> game logic)

**Review by mrakofff — CHANGES_REQUESTED** [review]

**Review by mrakofff — APPROVED** [review]

**Review by andreikaptsiuh — APPROVED** [review]

**Review by Praddby — APPROVED** [review]

**Pull request merged**